### ISSUE TITLE
`unnecessary_nullable_for_final_variable_declarations` should have tests with patterns

### ISSUE THREAD

**Issue by srawlins** [dart-lang/linter#4181]
Child issue of https://github.com/dart-lang/linter/issues/3624

I think there should be a test case like:

```dart
final (List<int>? a, List<int?> b, num? c) = ([], [], 1);
```

in which the lint rule would report that all three of the types are unnecessarily nullable.
